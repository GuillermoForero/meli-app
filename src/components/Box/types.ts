export interface BoxProps {
  width?: string;
  height?: string;
  minHeight?: string;
  justifyContent?: string;
  alignItems?: string;
  alignSelf?: string;
  flexDirection?: string;
  padding?: string;
  margin?: string;
  backgroundColor?: string;
}
