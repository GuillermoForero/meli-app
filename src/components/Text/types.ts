export type TextDecoration =
  | 'underline'
  | 'line-through'
  | 'overline';

export type TextAlign =
  | 'left'
  | 'center'
  | 'right';

export interface TextProps {
  color?: string;
  textAlign?: TextAlign;
  fontSize?: string;
  fontWeight?: string;
  margin?: string;
  className?: string;
  txtDecoration?: TextDecoration;
}
