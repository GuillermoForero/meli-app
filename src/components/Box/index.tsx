import styled from 'styled-components';

import { BoxProps } from './types';
//I created this reusable component because for SEO it would not affect much, due to the behavior of the website itself and the react
const Box = styled.div<BoxProps>`
  min-height: ${({ minHeight }) => (minHeight ? minHeight : false)};
  display: flex;
  justify-content: ${({ justifyContent }) => justifyContent || 'flex-start'};
  align-items: ${({ alignItems }) => alignItems || 'flex-start'};
  align-self: ${({ alignSelf }) => alignSelf || false};
  flex-direction: ${({ flexDirection }) => flexDirection || 'row'};
  box-sizing: border-box;
  height: ${({ height }) => height || 'auto'};
  width: ${({ width }) => width || '100%'};
  padding: ${({ padding }) => padding || '0'};
  margin: ${({ margin }) => margin || '0'};
`;

export default Box;
