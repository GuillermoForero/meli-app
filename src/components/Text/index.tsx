import styled from 'styled-components';
import { TextProps } from 'components/Text/types';
import { COLORS } from 'consts';
//I created this reusable component because for SEO it would not affect much, due to the behavior of the website itself and the react
const Text = styled.p.attrs((props: TextProps) => ({
  ...props,
  color: props.color || COLORS.GRAY_20,
}))`
  /* DEFAULT DEFINITIONS */
  font-family: 'Montserrat', sans-serif;
  margin: ${({ margin }) => margin? margin: '0'};
  text-decoration: ${({ txtDecoration }) =>
    txtDecoration ? txtDecoration : 'none'};
  font-size: ${({ fontSize }) => fontSize? fontSize: '12px'};
  font-weight: ${({ fontWeight }) => fontWeight? fontWeight: '400'};
  text-align: ${({ textAlign }) => textAlign? textAlign: 'left'};
  color: ${({ color }) => color};
  /* WORD-BREAKING DEFINITIONS */
  white-space: pre-wrap;
  white-space: -moz-pre-wrap;
  white-space: -pre-wrap;
  white-space: -o-pre-wrap;
  word-wrap: break-word;
  -ms-word-break: break-all;
  word-break: break-all;
  /* Non standard for WebKit */
  word-break: break-word;
`;

export default Text;
