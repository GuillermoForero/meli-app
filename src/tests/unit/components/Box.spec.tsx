import React from 'react';
import { render } from 'tests/support/customRender';
import Box from 'components/Box';
import Text from 'components/Text';

describe('[Box]', () => {
  it('Renders default snapshot', () => {
    const { container, getByText } = render(<Box>
      <Text>text example1</Text>
      <Text>text example2</Text>
    </Box>);
    expect(getByText('text example1')).toBeDefined();
    expect(getByText('text example2')).toBeDefined();
    expect(container).toMatchSnapshot();
  });
});
