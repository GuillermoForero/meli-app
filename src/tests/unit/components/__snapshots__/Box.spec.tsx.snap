// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Box] Renders default snapshot 1`] = `
<div>
  <div
    class="sc-bdfBwQ itqDgf"
  >
    <p
      class="sc-gsTCUz yjmFn"
    >
      text example1
    </p>
    <p
      class="sc-gsTCUz yjmFn"
    >
      text example2
    </p>
  </div>
</div>
`;
